packages/core/react/src/WalletProvider.tsx

```diff
@@ -23,285 +24,158 @@ export interface WalletProviderProps {
     localStorageKey?: string;
 }
 
-const initialState: {
-    wallet: Wallet | null;
-    adapter: Adapter | null;
-    publicKey: PublicKey | null;
-    connected: boolean;
-} = {
-    wallet: null,
-    adapter: null,
-    publicKey: null,
-    connected: false,
-};
-
-export const WalletProvider: FC<WalletProviderProps> = ({
-    children,
-    wallets: adapters,
-    autoConnect = false,
-    onError,
-    localStorageKey = 'walletName',
-}) => {
-    const [name, setName] = useLocalStorage<WalletName | null>(localStorageKey, null);
-    const [{ wallet, adapter, publicKey, connected }, setState] = useState(initialState);
-    const readyState = adapter?.readyState || WalletReadyState.Unsupported;
-    const [connecting, setConnecting] = useState(false);
-    const [disconnecting, setDisconnecting] = useState(false);
-    const isConnecting = useRef(false);
-    const isDisconnecting = useRef(false);
-    const isUnloading = useRef(false);
+let _userAgent: string | null;
+function getUserAgent() {
+    if (_userAgent === undefined) {
+        _userAgent = globalThis.navigator?.userAgent ?? null;
+    }
+    return _userAgent;
+}
 
-    // Wrap adapters to conform to the `Wallet` interface
-    const [wallets, setWallets] = useState(() =>
-        adapters.map((adapter) => ({
-            adapter,
-            readyState: adapter.readyState,
-        }))
-    );
+function getInferredClusterFromEndpoint(endpoint?: string): Cluster {
+    if (endpoint == null) {
+        return 'mainnet-beta';
+    }
```

Comment:
nit: can probably just do `!endpoint` in the odd case an empty string is provided.

Reply:
Done! I also moved all of this code into a module and redid the tests with a mock of that module.